### PR TITLE
Re-translate BLAS/LAPACK/ARPACK

### PR DESCRIPTION
**Pull request by szhorvat**
This re-translates BLAS/LAPACK/ARPACK with the latest f2c and updates to BLAS 3.12.0 and ARPACK-NG 3.7.0.

The ARPACK update alleviates problems like one reported at https://igraph.discourse.group/t/eigenvector-centrality-is-equal-to-0-for-all-nodes-even-in-a-strongly-connected-network/1898

Some unused f2c files are removed. A complete copy of f2c wasn't present in igraph anyway.

The thread-safety patch (mainly for ARPACK) is updated programmatically by replacing `static` with `IGRAPH_F77_SAVE` in all instances when the variable name does not start with `c_`, which I presume denotes a constant. A blanket replacement is not possible without compilation errors about constant initializers.  This approach probably introduces some unnecessary `IGRAPH_F77_SAVE` (which is a thread-local specifier) but better safe than sorry.

### CODE REVIEW

**Comment by szhorvat** [comment]
@ntamas I think this is good now. I am hoping that it will make ARPACK-based functions more robust.

The Windows issue was due to not patching this:

https://github.com/igraph/igraph/blob/e0dbe374e30adc654ad55a50d5aaf47ebc753908/tools/lapack/getlapack.sh#L141-L144

How did you do this last time? I just did it by a lot of manual find and replacing.

**Comment by szhorvat** [comment]
This PR significantly improves #1641.

File: vendor/lapack/arscnd.c [hunk 1]
```diff
@@ -28,15 +24,16 @@
     Purpose   
     =======   
 
-    SECOND returns the user time for a process in seconds.   
-    This version gets the time from the system function ETIME. */
+    SECOND returns the user time for a process in arscnds.
```
Comment (ntamas) [inline vendor/lapack/arscnd.c:27]: This replacement is probably unintended.

**Review by ntamas — APPROVED** [review]
Looks good to me but I had to admit that I did not check all the changes in the generated code thoroughly - I just assumed that `f2c` is doing its job correctly.

**Pull request merged**